src/machine/image-dbus.c

```diff
@@ -127,10 +127,25 @@ int bus_image_method_rename(
         if (r == 0)
                 return 1; /* Will call us back */
 
+        /* The image is cached with its name, hence it is necessary to remove from the cache before renaming. */
+        Image *removed = hashmap_remove(m->image_cache, image->name);
```

Comment:
Hmm, the memory management here is rather confusing? If image_rename fails this is leaked?